validation-test/Sema/rdar34333874.swift

```diff
@@ -0,0 +1,24 @@
+// RUN: rm -rf %t
+// RUN: mkdir -p %t
+// RUN: %target-build-swift %s -o %t/a.out
+// RUN: %target-run %t/a.out
+
+// REQUIRES: executable_test
+
+class C {
+  var a: Int = 0
+  var b: Int = 0
+}
+
+func foo<T>(_ item: T, update: (inout T) throws -> Void) rethrows -> T {
+   var this = item
+   try update(&this)
+   return this
+}
+
+let rdar34333874 = foo(C()) {
+  $0.a = 42
```

Comment:
Is this really testing the same thing as the reported bug? This actually is mutating the input whereas the bug report wasn't.

Reply:
Example from the rdar does modify fields of `$0`

Reply:
The actual original code did mutate input inside the closure. I'd suggest testing both cases to be on the safe side.

Reply:
Sure, sounds good, I'll add one which doesn't mutate the fields.